api/app/Controllers/Http/ChunkedUploadController.js

```diff
@@ -0,0 +1,99 @@
+"use strict";
+const Env = use("Env");
+const path = require("path");
+const fs = require("fs");
+const os = require("os");
+
+const TIMEOUT = 10*60000; // Timeout in milliseconds (30 seconds)
+
+class ChunkedUploadController {
+  deletePartialStoredChunks(totalChunks) {
+    const uploadPath = os.tmpdir();
+    for (let i = 0; i < totalChunks; i++) {
+      const tmpFilePath = path.join(uploadPath, `tempfile${index}.bin`);
+      fs.unlinkSync(tmpFilePath); // Remove individual chunk files
+    }
+  }
+
+  async upload({ request, response, session }) {
+    const { chunk } = request.post();
+    const { chunkNumber, totalChunks, filename } = request.only([
+      "chunkNumber",
+      "totalChunks",
+      "filename",
+    ]);
+    const file = chunk;
+    const chunkKey = `${filename}_${totalChunks}`;
+    const chunks = session.get(chunkKey, []);
+    chunks[chunkNumber] = Date.now();
+    session.put(chunkKey, chunks);
+
+    const timeoutId = setTimeout(() => {
```

Comment:
I would not set a timeout here which basically means you register a callback that is executed the timeout period later for **each upload** call which might overload backend in case of an upload with many chunks and think of concurrent users. You can omit this here and simply call the function deletePartialStoredChunks at the end of the function (or in case of error where you do that already) and the function scans the temporary directory for files, retrieves fstat on last chunk file only. You could work with glob function here. You can approach Hemant, he has been using that somewhere else.